vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -651,3 +669,31 @@ func (s *GenericAPIServer) Eventf(eventType, reason, messageFmt string, args ...
 		klog.Warningf("failed to create event %s/%s: %v", e.Namespace, e.Name, err)
 	}
 }
+
+// terminationLoggingListener wraps the given listener and logs new connections
+// after the stopCh has been closed, i.e. when termination has begun.
+type terminationLoggingListener struct {
+	net.Listener
+	stopCh, lateStopCh <-chan struct{}
+}
+
+func (l *terminationLoggingListener) Accept() (net.Conn, error) {
+	c, err := l.Listener.Accept()
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case <-l.lateStopCh:
+		klog.Warningf("Accepted new connection from %s very late in the termination process, possibly a hint for a broken load balancer setup.", c.RemoteAddr().String())
```

Comment:
do we want to specify what `very late` actually means? i.e. 80% of time has already elapsed

Reply:
yes

Reply:
done

Reply:
done